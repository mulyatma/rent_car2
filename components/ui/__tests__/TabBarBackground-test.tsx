import React from 'react';
import renderer from 'react-test-renderer';
import { useBottomTabBarHeight } from '@react-navigation/bottom-tabs';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import BlurTabBarBackground, { useBottomTabOverflow } from '../TabBarBackground';

// Mocking dependencies
jest.mock('@react-navigation/bottom-tabs', () => ({
  useBottomTabBarHeight: jest.fn(),
}));

jest.mock('react-native-safe-area-context', () => ({
  useSafeAreaInsets: jest.fn(),
}));

jest.mock('expo-blur', () => ({
  BlurView: 'BlurView',
}));

describe('BlurTabBarBackground', () => {
  it('renders the BlurView with correct props', () => {
    const tree = renderer.create(<BlurTabBarBackground />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('useBottomTabOverflow', () => {
  it('calculates the bottom tab overflow correctly', () => {
    // Mock values
    const mockTabHeight = 50;
    const mockBottomInset = 20;

    (useBottomTabBarHeight as jest.Mock).mockReturnValue(mockTabHeight);
    (useSafeAreaInsets as jest.Mock).mockReturnValue({ bottom: mockBottomInset });

    // Call the hook
    const overflow = useBottomTabOverflow();

    // Verify the result
    expect(overflow).toBe(mockTabHeight - mockBottomInset);
  });

  it('returns the correct value when bottom inset is zero', () => {
    (useBottomTabBarHeight as jest.Mock).mockReturnValue(60);
    (useSafeAreaInsets as jest.Mock).mockReturnValue({ bottom: 0 });

    const overflow = useBottomTabOverflow();
    expect(overflow).toBe(60);
  });
});
