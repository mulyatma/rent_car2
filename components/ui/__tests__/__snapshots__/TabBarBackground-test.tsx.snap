// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlurTabBarBackground renders the BlurView with correct props 1`] = `
<BlurView
  intensity={100}
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  tint="systemChromeMaterial"
/>
`;
